addons/binding/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusNetworkHandler.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cbus.handler;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.cbus.CBusBindingConstants;
+import org.openhab.binding.cbus.internal.cgate.CGateException;
+import org.openhab.binding.cbus.internal.cgate.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CBusNetworkHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Linton - Initial contribution
+ */
+public class CBusNetworkHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+    private CBusCGateHandler bridgeHandler;
+    private Network network;
+
+    public CBusNetworkHandler(Bridge thing) {
+        super(thing);
+        if (thing == null) {
+            logger.error("Required bridge not defined for device.");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+    }
+
+    @Override
+    public void initialize() {
+        getCBusCGateHandler();
+        if (!bridgeHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+        String networkID = getConfig().get(CBusBindingConstants.PROPERTY_ID).toString();
+        String project = getConfig().get(CBusBindingConstants.PROPERTY_PROJECT).toString();
+        try {
+            network = (Network) getCBusCGateHandler().getCGateSession()
+                    .getCGateObject("//" + project + "/" + networkID);
+        } catch (CGateException e) {
+            logger.error("Cannot load C-Bus network {}", networkID, e);
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+        updateStatus();
+        // now also re-initialize all group handlers
+        for (Thing thing : getThing().getThings()) {
+            ThingHandler handler = thing.getHandler();
+            if (handler != null) {
+                handler.initialize();
```

Comment:
handlers are not supposed to call initialize() on others - this is only done by the framework.
Note that with a recent refactoring, it is made sure that the bridge handler is always initialized before any thing handler, so there should not be a need for a "re-initialization" anymore.